appserver/web/admin/src/main/java/fish/payara/grizzly/config/admin/cli/RestartHttpListenersCommand.java

```diff
@@ -0,0 +1,141 @@
+ *  enclosed by brackets [] replaced by your own identifying information:
+ *  "Portions Copyright [year] [name of copyright owner]"
+ *
+ *  Contributor(s):
+ *  If you wish your version of this file to be governed by only the CDDL or
+ *  only the GPL Version 2, indicate your decision by adding "[Contributor]
+ *  elects to include this software in this distribution under the [CDDL or GPL
+ *  Version 2] license."  If you don't indicate a single choice of license, a
+ *  recipient has the option to distribute your version of this file under
+ *  either the CDDL, the GPL Version 2 or to extend the choice of license to
+ *  its licensees as provided above.  However, if you add GPL Version 2 code
+ *  and therefore, elected the GPL Version 2 license, then the option applies
+ *  only if the new code is made subject to such option by the copyright
+ *  holder.
+ */
+package fish.payara.grizzly.config.admin.cli;
+
+import java.text.MessageFormat;
+import java.util.concurrent.TimeUnit;
+
+import javax.inject.Inject;
+
+import org.glassfish.api.ActionReport;
+import org.glassfish.api.ActionReport.ExitCode;
+import org.glassfish.api.Param;
+import org.glassfish.api.admin.AdminCommand;
+import org.glassfish.api.admin.AdminCommandContext;
+import org.glassfish.api.admin.ExecuteOn;
+import org.glassfish.api.admin.FailurePolicy;
+import org.glassfish.api.admin.ParameterMap;
+import org.glassfish.api.admin.RestEndpoint;
+import org.glassfish.api.admin.RestEndpoints;
+import org.glassfish.api.admin.RuntimeType;
+import org.glassfish.config.support.CommandTarget;
+import org.glassfish.config.support.TargetType;
+import org.glassfish.grizzly.config.dom.NetworkListener;
+import org.glassfish.grizzly.config.dom.NetworkListeners;
+import org.glassfish.hk2.api.PerLookup;
+import org.glassfish.hk2.api.ServiceLocator;
+import org.glassfish.internal.api.Target;
+import org.jvnet.hk2.annotations.Service;
+
+import com.sun.enterprise.admin.util.ClusterOperationUtil;
+import com.sun.enterprise.config.serverbeans.Config;
+import com.sun.enterprise.config.serverbeans.Domain;
+import com.sun.enterprise.util.SystemPropertyConstants;
+import com.sun.enterprise.v3.services.impl.GrizzlyService;
+
+@ExecuteOn({RuntimeType.DAS, RuntimeType.INSTANCE})
+@TargetType(value = {CommandTarget.DAS, CommandTarget.STANDALONE_INSTANCE, CommandTarget.CLUSTER, CommandTarget.CONFIG, CommandTarget.DEPLOYMENT_GROUP})
+@Service(name = "restart-http-listeners")
+@PerLookup
+@RestEndpoints({
+    @RestEndpoint(configBean = NetworkListeners.class,
+            opType = RestEndpoint.OpType.POST,
+            description = "Restarts all HTTP listeners")
+})
+public class RestartHttpListenersCommand implements AdminCommand {
+
+    @Inject
+    private Domain domain;
+
+    @Inject
+    private ServiceLocator locator;
+
+    @Inject
+    private Target targetUtil;
+
+    @Inject
+    private GrizzlyService service;
+
+    @Param(name = "target", optional = true)
```

Comment:
```suggestion
    @Param(name = "target", optional = true, primary = true)
```
Save people writing out _--target_ - I think it's fairly safe to assume if someone types the below they mean an instance
`asadmin restart-http-listeners Flabbergasted-Blobfish`